geoips/plugins/modules/output_checkers/utils/compare_outputs.py

```diff
@@ -541,35 +135,42 @@ def test_product(output_product, compare_product, goodcomps, badcomps, compare_s
         Raised when current output product does not have an associated
         comparison test defined.
     """
+    from geoips.plugins.modules.output_checkers import (
+        geotiff_checker,
+        image_checker,
+        netcdf_checker,
+        text_checker,
+    )
+
     matched_one = False
-    if is_image(output_product):
+    if image_checker.correct_type(output_product):
         matched_one = True
         compare_strings += ["IMAGE "]
-        if images_match(output_product, compare_product):
+        if image_checker.images_match(output_product, compare_product):
             goodcomps += ["IMAGE {0}".format(output_product)]
         else:
             badcomps += ["IMAGE {0}".format(output_product)]
 
-    if is_geotiff(output_product):
+    if geotiff_checker.correct_type(output_product):
         matched_one = True
         compare_strings += ["GEOTIFF "]
-        if geotiffs_match(output_product, compare_product):
+        if geotiff_checker.geotiffs_match(output_product, compare_product):
             goodcomps += ["GEOTIFF {0}".format(output_product)]
         else:
             badcomps += ["GEOTIFF {0}".format(output_product)]
 
-    if is_text(output_product):
+    if text_checker.correct_type(output_product):
         matched_one = True
         compare_strings += ["TEXT "]
-        if text_match(output_product, compare_product):
+        if text_checker.text_match(output_product, compare_product):
             goodcomps += ["TEXT {0}".format(output_product)]
         else:
             badcomps += ["TEXT {0}".format(output_product)]
 
-    if is_geoips_netcdf(output_product):
+    if netcdf_checker.correct_type(output_product):
         matched_one = True
         compare_strings += ["GEOIPS NETCDF "]
-        if geoips_netcdf_match(output_product, compare_product):
+        if netcdf_checker.geoips_netcdf_match(output_product, compare_product):
             goodcomps += ["GEOIPS NETCDF {0}".format(output_product)]
         else:
             badcomps += ["GEOIPS NETCDF {0}".format(output_product)]
```

Comment:
I think this would all be better directly in the interface, so the actual output checkers are standalone.  Ie, from within the interface, atuomatically identify all of the output checkers, then find call "correct_type" for each one until you find the right one.

We should probably eventually have a way to explicitly specify the output checker type, but for now looping through all of them until we find the right one is probably fine (that's what it's doing now)

Reply:
This has been addressed.  Resolving.